compiler/rustc_parse/src/lib.rs

```diff
@@ -443,12 +443,21 @@ pub fn tokenstream_probably_equal_for_proc_macro(
     // that are produced as a result of nonterminal expansion.
     let mut t1 = first.trees().filter(semantic_tree).flat_map(expand_nt).flat_map(break_tokens);
     let mut t2 = other.trees().filter(semantic_tree).flat_map(expand_nt).flat_map(break_tokens);
-    for (t1, t2) in t1.by_ref().zip(t2.by_ref()) {
-        if !tokentree_probably_equal_for_proc_macro(&t1, &t2, sess) {
-            return false;
+
+    // We cannot use 'zip()' here, as it can terminate early while still
+    // consuming both iterators (in the case where the right-hand iterator
+    // has one less item).
+    loop {
+        match (t1.next(), t2.next()) {
+            (Some(tree1), Some(tree2)) => {
+                if !tokentree_probably_equal_for_proc_macro(&tree1, &tree2, sess) {
+                    return false;
+                }
+            }
+            (None, None) => return true,
+            _ => return false,
         }
     }
```

Comment:
```suggestion
    t1.eq_by(t2, |tree1, tree2| tokentree_probably_equal_for_proc_macro(&tree1, &tree2, sess))
```
But would require #![feature(iter_order_by)] at the top of the file.

Reply:
I didn't know that that method existed - thanks!

Reply:
I didn't know either until I looked at your PR and my brain told me it was a method that had to exist, so I checked to see if it did, and it did (on nightly)!